x-pack/plugins/security/public/management/role_mappings/edit_role_mapping/edit_role_mapping_page.tsx

```diff
@@ -211,6 +214,49 @@ export class EditRoleMappingPage extends Component<Props, State> {
     );
   };
 
+  private checkEmptyAnyAllMappings = (obj: RoleMappingRule) => {
+    const arrToCheck: RoleMappingRule[] = [obj];
+
+    while (arrToCheck.length > 0) {
+      const currentObj = arrToCheck.pop();
+      if (typeof currentObj === 'object' && currentObj !== null) {
+        for (const key in currentObj) {
+          if (currentObj.hasOwnProperty(key)) {
+            const value: unknown = currentObj[key as keyof RoleMappingRule];
```

Comment:
nit 
I think we might be able to extract value 
```typescript
type ValueOf<T> = T[keyof T];

export type RoleMappingRuleValue =
  | ValueOf<RoleMappingAnyRule>
  | ValueOf<RoleMappingAllRule>
  | ValueOf<RoleMappingExceptRule>
  | ValueOf<RoleMappingFieldRule>;
```

```suggestion
            const value: RoleMappingRuleValue = currentObj[key as keyof RoleMappingRule];
```

Reply:
I removed the `const value...` line entirely and replaced with the typeguards.